src/main/java/baritone/utils/schematic/format/defaults/LitematicaSchematic.java

```diff
@@ -0,0 +1,345 @@
+        IBlockState blockState = block.getDefaultState();
+
+        for (Object key : properties.getKeySet().toArray()) {
+            IProperty<?> property = block.getBlockState().getProperty((String) key);
+            String propertyValue = properties.getString((String) key);
+            if (property != null) {
+                blockState = setPropertyValue(blockState, property, propertyValue);
+            }
+        }
+        return blockState;
+    }
+
+    /**
+     * @author Emerson
+     */
+    private static <T extends Comparable<T>> IBlockState setPropertyValue(IBlockState state, IProperty<T> property, String value) {
+        Optional<T> parsed = property.parseValue(value).toJavaUtil();
+        if (parsed.isPresent()) {
+            return state.withProperty(property, parsed.get());
+        } else {
+            throw new IllegalArgumentException("Invalid value for property " + property);
+        }
+    }
+
+    /**
+     * @param amountOfBlockTypes amount of block types in the schematic.
+     * @return amount of bits used to encode a block.
+     */
+    private static int getBitsPerBlock(int amountOfBlockTypes) {
+        return (int) Math.floor((Math.log(amountOfBlockTypes)) / Math.log(2)) + 1;
+    }
+
+    /**
+     * Calculates the volume of the subregion. As size can be a negative value we take the absolute value of the
+     * multiplication as the volume still holds a positive amount of blocks.
+     *
+     * @return the volume of the subregion.
+     */
+    private static long getVolume(NBTTagCompound nbt, String subReg) {
+        return Math.abs(
+                nbt.getCompoundTag("Regions").getCompoundTag(subReg).getCompoundTag("Size").getInteger("x") *
+                        nbt.getCompoundTag("Regions").getCompoundTag(subReg).getCompoundTag("Size").getInteger("y") *
+                        nbt.getCompoundTag("Regions").getCompoundTag(subReg).getCompoundTag("Size").getInteger("z"));
+    }
+
+    /**
+     * @return array of Long values.
+     */
+    private static long[] getBlockStates(NBTTagCompound nbt, String subReg) {
+        return ((INBTTagLongArray) nbt.getCompoundTag("Regions").getCompoundTag(subReg).getTag("BlockStates")).getLongArray();
+    }
+
+    /**
+     * Subregion don't have to be the same size as the enclosing size of the schematic. If they are smaller we check here if the current block is part of the subregion.
+     *
+     * @param x coord of the block relative to the minimum corner.
+     * @param y coord of the block relative to the minimum corner.
+     * @param z coord of the block relative to the minimum corner.
+     * @return if the current block is part of the subregion.
+     */
+    private static boolean inSubregion(NBTTagCompound nbt, String subReg, int x, int y, int z) {
+        return
+                x < Math.abs(nbt.getCompoundTag("Regions").getCompoundTag(subReg).getCompoundTag("Size").getInteger("x")) &&
```

Comment:
does this also need to check if they are negative? should this have a `x >=0 && y >= 0 && z >=0 && ` at the front?

Reply:
no the xyz values come from a counter in a for loop that counts from 0 up to schemsize-1

Reply:
ok even if it doesn't need the check please add it for correctness sake (it is definitely incorrect that `inSubregion(-1000000, -1000000, -1000000)` would return true)